templates-v7/libraries/jersey3/JSON.mustache

```diff
@@ -28,10 +28,10 @@ import jakarta.ws.rs.ext.ContextResolver;
 public class JSON implements ContextResolver<ObjectMapper> {
   private static ObjectMapper mapper;
 
-  public JSON() {
+  private JSON() {
     mapper = new ObjectMapper();
     mapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
-    JsonMapper.builder().configure(MapperFeature.ALLOW_COERCION_OF_SCALARS, false);
+    mapper.configure(MapperFeature.ALLOW_COERCION_OF_SCALARS, true);
```

Comment:
`.configure(MapperFeature.ALLOW_COERCION_OF_SCALARS, false)` This changes the json behavior from a more `strict type of matching` to a more `flexible` matching approach, could you add the reason for this specific change?